### PR TITLE
adding pc name to loggly messages

### PR DESCRIPTION
**Pull request by pnyengan**
Please review and merge

### CODE REVIEW

File: scraper.go [hunk 1]
```diff
@@ -153,8 +156,7 @@ func NewScraper(hostname, statefile, metalogfile string) *Scraper {
 	s := &Scraper{}
 	if hostname != "" {
 		s.Hostname = hostname
-	} else {
-		s.Hostname, _ = os.Hostname()
+		s.PCname,_ = os.Hostname()
```
Comment (nicovanniekerk) [inline scraper.go]: if i understand this correctly, if hostname is empty we will not be seeing the host or the PC name in the loggly message, please revisit this and add the else back to add the pcsname if hostname if empty.

File: scraper.go [hunk 2]
```diff
@@ -106,12 +107,13 @@ type logglyJsonMsg struct {
 	JavaClass        string  `json:"java_class,omitempty"`
 }
 
-func (m *LogMsg) toLogglyJson(hostname string, source string, target *json.Encoder) error {
+func (m *LogMsg) toLogglyJson(hostname string, pcname, source string, target *json.Encoder) error {
```
Comment (nicovanniekerk) [inline scraper.go]: pcname no type?

**Comment by pnyengan** [comment]
@nicovanniekerk  changes made please have a look

**Comment by nicovanniekerk** [comment]
LGTM, @bradmunz79 @Na-iemDollieIMQS ?

**Comment by bmharper** [comment]
This commit can be much simpler.
You don't need to push the "pcname" logic out into so many parts of the code. Just keep it inside the LogScraper constructor. It's always os.Hostname, so just read os.Hostname inside the LogScraper constructor. It doesn't need to affect all the other parts of the code, such as getHostname, etc.
Also, is PCName really the best term for this?
Perhaps "OwnHostname"?
PC = Personal Computer. Just seems a bit anachronistic to me.

**Comment by bmharper** [comment]
Also, gofmt!!!

**Comment by Na-iemDollieIMQS** [comment]
@bmharper The original getHostName() was (potentially) calling os.HostName(), so that it could use os.HostName for the hostname variable if the real hostname is not available (e.g. no hostname file).
Now we're simply returning os.HostName() as the pcName as well, instead of having to make another system call in the constructor.

I suppose we could first call os.HostName in the constructor, assign it to pcName, and then pass that as a parameter to getHostName.
However, (as I understand it) the responsibility of the getHostName function is to supply us with some sort of identity of the host. It might then be better keep all implementation code to obtain the "host identity details" inside getHostName. Actually, thinking about it some more, do you think we should rename getHostName to getHostIdentity?

**Comment by bmharper** [comment]
@Na-iemDollieIMQS @pnyengan The reason I originally placed `getHostname` inside main.go, is because I don't want to place such deep linking with other IMQS infrastructure, into the `logscraper` package. So I think that should remain so. The words could perhaps be a little different, and I think that `getHostname` should now be altered so that instead of returning `os.Hostname` as a fallback, it just returns an empty string.
The call to `os.Hostname` should remain inside `NewScraper`, but it should be assigned to a new variable -- perhaps `OSHostName`. The pair should always be transmitted, even if the 'hostname' one is blank.

**Comment by bmharper** [comment]
Stick to the same name everywhere. If it's OSHostname, then use that string everywhere - use it in the JSON, and in every variable name. If you choose to use OwnHostname, then use that everywhere. Either name is fine, but be consistent.

**Comment by pnyengan** [comment]
@bmharper please review the code changes.

File: scraper.go [hunk 3]
```diff
@@ -106,12 +107,13 @@ type logglyJsonMsg struct {
 	JavaClass        string  `json:"java_class,omitempty"`
 }
 
-func (m *LogMsg) toLogglyJson(hostname string, source string, target *json.Encoder) error {
+func (m *LogMsg) toLogglyJson(hostname string, OwnHostname string, source string, target *json.Encoder) error {
```
Comment (bmharper) [inline scraper.go]: local variables start with lower case.
Also -- there is still the variable called "PC" in here.

Comment (bmharper) [reply scraper.go]: All of our other JSON fields are lower_case, so stick to that for the JSON message.

File: scraper.go [hunk 4]
```diff
@@ -143,19 +145,17 @@ type stateJson struct {
 type Scraper struct {
 	Sources       []*LogSource
 	Hostname      string
+	PCname        string
```
Comment (bmharper) [inline scraper.go]: Same name everywhere. Not PCname. Stick to ownhostname. In this case, to be specific, it should be OwnHostname.

Comment (pnyengan) [reply scraper.go]: removed PCname and Used OwnHostName everywhere

**Pull request merged**